source/MaterialXGenShader/TypeDesc.cpp

```diff
@@ -47,7 +47,7 @@ const TypeDesc* TypeDesc::registerType(const string& name, unsigned char basetyp
         throw ExceptionShaderGenError("A type with name '" + name + "' is already registered");
     }
 
-    TypeDescPtr ptr = TypeDescPtr(new TypeDesc(name, basetype, semantic, size, editable, channelMapping));
+    TypeDescPtr ptr = TypeDescPtr(new TypeDesc(name, basetype, semantic, size, editable, channelMapping), TypeDesc::destroy);
```

Comment:
Given the current behavior of TypeDesc::registerType, I see the logic behind implementing Python bindings in this way, but I'm wondering if perhaps it would be better to pass type descriptions consistently by TypeDescPtr.  Adding @bernardkwok for his thoughts as well.

Reply:
To keep type comparisons as efficient as possible, another option would be to define TypeDescPtr as a _raw pointer_, removing smart pointers entirely from the management of TypeDesc object lifetimes.  This seems more in keeping with the spirit of the current design, where all TypeDesc objects are allocated during static initialization, and are deleted during static destruction.